docs/rfc/rfc-023-semi-permanent-testnet.md

```diff
@@ -0,0 +1,206 @@
+The intent of the testnet being discussed in this RFC is to exist indefinitely;
+but there is a practical understanding that there will be testnet instances
+which will be retired due to being associated with a no-longer-maintained release of Tendermint.
+Alternatively, we may discover a good reason to destroy and recreate the `main` testnet
+(such as if we build a new application for running the testnet).
+
+### Testnet MVP
+
+The minimum viable testnet meets this set of features:
+
+- The testnet self-updates following a new commit pushed to Tendermint's `main` branch on GitHub
+  (there are some omitted steps here, such as CI building appropriate binaries and
+  somehow notifying the testnet that a new build is available)
+- The testnet runs the Tendermint KV store for MVP
+- The testnet operators are notified if any node's process exits for any reason other than
+  a restart for a new binary
+- The testnet operators are notified if any node stops updating blocks,
+  and by extension if a chain halt occurs
+- The testnet has a minimum of 1 full node and 3 validators
+- The testnet has a reasonably low, constant throughput of transactions -- say 30 tx/min --
+  and the testnet operators are notified if that throughput drops below 75% of target
+  sustained over 5 minutes
+- The testnet only needs to run in a single datacenter/cloud-region for MVP,
+  i.e. running in multiple datacenters is out of scope for MVP
+- The testnet is running directly on VMs or compute instances;
+  while Kubernetes or other orchestration frameworks may offer many significant advantages,
+  the Tendermint engineers should not be required to learn those tools in order to
+  perform basic debugging
+
+### Testnet medium-term goals
+
+The medium-term goals are intended to be achievable within the 6-12 month time range
+following the launch of MVP.
+These goals could realistically be roadmapped following the launch of the MVP testnet.
+
+- The `main` testnet has more than 20 nodes (completely arbitrary -- 5x more than 1+3 at MVP)
+- In addition to the `main` testnet,
+  there is at least one testnet associated with one release branch
+- The testnet no longer is simply running the Tendermint KV store;
+  now it is built on a more complex, custom application
+  that deliberately exercises a greater portion of the Tendermint stack
+- Each testnet is spread across at least two cloud providers,
+  in order to communicate over a network more closely resembling use of Tendermint in "real" chains
+- The node updates have some "jitter",
+  with some nodes updating immediately when a new build is available,
+  and others delaying up to perhaps 30-60 minutes
+- The team has published some form of dashboards that have served well for debugging,
+  which external parties can copy/modify to their needs
+- The team has produced a reference model of a log aggregation stack that external parties can use
```

Comment:
I like both of these a lot, but they are basically their own projects. If we're going to do this, I think we should spend a bit of time on these dotting our i's and crossing our t's. A MVP of these dashboards will be probably a bit sloppy and ramping up to something releasable will take some polishing. My concern, ultimately, is around how we position any such stack/dashboard. If we communicate that this is the Tendermint stack, we may come to own its use in the while in a way we don't intend. I think we'd need to make sure to position any such resources as 'things that we use but are very much on the user to tinker with'.